src/main/java/avd/inf/jdm/rentmycar/domain/Booking.java

```diff
@@ -39,4 +43,9 @@ public Booking(Offer offer, User customer) {
         this.status = BookingStatus.PENDING;
     }
 
+    public Booking(Offer offer, Ride ride, User customer) {
```

Comment:
I don't think `Ride` should be in the constructor, since it does not exist yet when a `Booking` is created. I would suggest we create a `startRide(Booking booking)` method on the `BookingService` Class that instantiates a `Ride` and sets it to the `Booking`. That method would also set the `BookingStatus` of the` Booking` to `CARPICKEDUP` (or whatever name I gave it 😅 )

Reply:
Great! Thanks for this Roy. I've changed this.